geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/protobufs/TupleProtoBuf.scala

```diff
@@ -50,6 +74,17 @@ trait TupleProtoBuf {
         multibandTileProtoBufCodec.decode(message.tiles.get))
   }
 
+  implicit def tupleSpatialKeyImageProtoBufCodec =
+    new ProtoBufCodec[(SpatialKey, Array[Byte]), ProtoTuple] {
```

Comment:
You could save having to enumerate all the cases here by parameterizing the `K` to get signature like:

```
implicit def tupleSpatialKeyImageProtoBufCodec[K](implicit keyCodec: ProtoBufCodec[K]) =
  new ProtoBufCodec[(K, Array[Byte]), ProtoTuple] {
    def encode(tuple: (SpatialKey, Array[Byte])): ProtoTuple =
      ProtoTuple(
        spatialKey = Some(keyCodec.encode(tuple._1)),
        imageBytes = ByteString.copyFrom(tuple._2))

    def decode(message: ProtoTuple): (SpatialKey, Array[Byte]) = {
      ???
    }
  }  ...
```